src/audiomixerboard.cpp

```diff
@@ -1210,87 +1210,93 @@ void CAudioMixerBoard::ApplyNewConClientList ( CVector<CChannelInfo>& vecChanInf
 
         // search for channels with are already present and preserve their gain
         // setting, for all other channels reset gain
+
+        // get all channels which are in use/not in use. We use the array index of vecChanInfo
+        // if the fader is in use, else INVALID_INDEX to specify it is not in use
+        int iFaderNumber[MAX_NUM_CHANNELS];
+
         for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
         {
-            bool bFaderIsUsed = false;
+            iFaderNumber[i] = INVALID_INDEX;
+        }
 
-            for ( int j = 0; j < iNumConnectedClients; j++ )
+        for ( int i = 0; i < iNumConnectedClients; i++ )
+        {
+            iFaderNumber[vecChanInfo[i].iChanID] = i;
+        }
+
+        for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+        {
+            if ( iFaderNumber[i] != INVALID_INDEX )
```

Comment:
If we're refactoring the control flow anyway, I'd suggest moving the non-negated and shorter part first.

```suggestion
            if ( iFaderNumber[i] == INVALID_INDEX )
            {
                // fader unused
                continue;
            }
            // fader used, more logic
```

Reply:
There's an else to hide the fader. (See below).

But it can be swapped for sure

Reply:
Yes, sorry, I wasn't clear. I didn't mean to reduce it to the comments but thought of them as placeholders. :)

Reply:
Ok. Done. Please re-check

Reply:
Looks good, thanks.